### PR TITLE
chore(examples): switch to Script component on the google analytics example

### PR DESCRIPTION
**Pull request by smakosh**
## Documentation / Examples

- [x] Make sure the linting passes

Fixes https://github.com/vercel/next.js/pull/27674

### CODE REVIEW

**Review by styfle — COMMENTED** [review]
Thanks! We'll have @janicklas-ralph take a look soon to ensure this meets Google's recommendation.

**Review by leerob — APPROVED** [review]

**Review by leerob — CHANGES_REQUESTED** [review]
👍 Will wait for Google's recommendation!

**Comment by janicklas-ralph** [comment]
Hi @smakosh. I will update this PR when I've confirmed that this would be the best way to load GA. Thanks

File: examples/with-google-analytics/pages/_app.js [hunk 1]
```diff
@@ -14,7 +15,30 @@ const App = ({ Component, pageProps }) => {
     }
   }, [router.events])
```
Comment (ddzieduch) [inline examples/with-google-analytics/pages/_app.js:17]: Is this part still necessary?

Comment (smakosh) [reply examples/with-google-analytics/pages/_app.js:17]: This tracks page views when the route changes

File: examples/with-google-analytics/pages/_app.js [hunk 2]
```diff
@@ -14,7 +15,30 @@ const App = ({ Component, pageProps }) => {
...
+
+            gtag('config', '${gtag.GA_TRACKING_ID}', {
+              page_path: window.location.pathname,
+            });
+          `,
+        }}
+      />
+      <Component {...pageProps} />
+    </>
+  )
```
Comment (housseindjirdeh) [inline examples/with-google-analytics/pages/_app.js:41]: @janicklas-ralph @rgabs Our ESLint rule [doc](https://github.com/vercel/next.js/blob/canary/errors/next-script-for-ga.md#using-gtagjs) suggests to use `lazyOnLoad` when loading analytics via `gtag`. We need to make sure our messaging is aligned with the approach used in all the `examples/`.

Comment (smakosh) [reply examples/with-google-analytics/pages/_app.js:41]: Updated @housseindjirdeh 👍

Comment (pqminh) [reply examples/with-google-analytics/pages/_app.js:41]: afterInteractive (default): For scripts that can fetch and execute after the page is interactive, such as tag managers and analytics. These scripts are injected on the client-side and will run after hydration.

[Docs page](https://nextjs.org/docs/basic-features/script)
This make me confuse!
Should I use default strategy(afterInteractive) or lazyOnLoad?

Comment (smakosh) [reply examples/with-google-analytics/pages/_app.js:39]: Updated that doc as well on this PR

File: examples/with-google-analytics/pages/_app.js [hunk 3]
```diff
@@ -14,7 +15,28 @@ const App = ({ Component, pageProps }) => {
     }
   }, [router.events])
 
-  return <Component {...pageProps} />
+  return (
+    <>
+      {/* Global Site Tag (gtag.js) - Google Analytics */}
+      <Script
```
Comment (rgabs) [inline examples/with-google-analytics/pages/_app.js:21]: Would be nice to add the loading strategy to make it in sync with the documentation(even if afterInteractive is default)

File: examples/with-google-analytics/pages/_app.js (hunk 2 above)
Comment (rgabs) [reply examples/with-google-analytics/pages/_app.js:39]: Thanks @smakosh 👏

File: examples/with-google-analytics/pages/_app.js (hunk 3 above)
Comment (smakosh) [reply examples/with-google-analytics/pages/_app.js:21]: Makes sense

File: examples/with-google-analytics/pages/_app.js (hunk 2 above)
Comment (housseindjirdeh) [reply examples/with-google-analytics/pages/_app.js:41]: Thanks @smakosh, `afterInteractive` is the correct approach so appreciate you updating the documentation to make them aligned :)

**Review by housseindjirdeh — APPROVED** [review]
This LGTM!

CC @janicklas-ralph for his review

**Review by janicklas-ralph — APPROVED** [review]

**Review by styfle — APPROVED** [review]

**Review by leerob — APPROVED** [review]

**Review by ijjk — APPROVED** [review]

**Comment by TryingToImprove** [comment]
I just updated my page to use the same as the example, however it breaks `reportWebVitals` telling that `window.gtag is not a function`. Moving the script to `_document.tsx` fixes the issue.

**Comment by smakosh** [comment]
@TryingToImprove I'm using the same approach on the example here: https://github.com/smakosh/next-portfolio-dev/blob/main/src/pages/_app.tsx and it's working fine

**Comment by TryingToImprove** [comment]
> @TryingToImprove I'm using the same approach on the example here: https://github.com/smakosh/next-portfolio-dev/blob/main/src/pages/_app.tsx and it's working fine

You don't have `reportWebVitals` in your example..  (https://nextjs.org/docs/advanced-features/measuring-performance#sending-results-to-analytics)

**Comment by smakosh** [comment]
Because I'm using Next js analytics and Vercel handles that for me out of the box.

**Pull request merged**